Lois/src/server.js

```diff
@@ -0,0 +1,116 @@
+const bodyParser = require('body-parser');
+const express = require('express');
+const session = require('express-session');
+const bcrypt = require('bcrypt');
+
+const User = require('./user');
+
+const STATUS_USER_ERROR = 422;
+const BCRYPT_COST = 11;
+
+const server = express();
+// to enable parsing of json bodies for post requests
+server.use(bodyParser.json());
+server.use(session({
+  secret: 'e5SPiqsEtjexkTj3Xqovsjzq8ovjfgVDFMfUzSmJO21dtXs4re',
+  resave: true,
+  saveUninitialized: true,
+}));
+
+/* Sends the given err, a string or an object, to the client. Sets the status
+ * code appropriately. */
+const sendUserError = (err, res) => {
+  res.status(STATUS_USER_ERROR);
+  if (err && err.message) {
+    res.json({ message: err.message, stack: err.stack });
+  } else {
+    res.json({ error: err });
+  }
+};
+
+// TODO: implement routes
+const userAuthMiddleware = (req, res, next) => {
+  if (req.session.user === undefined) {
```

Comment:
Here you can just check `if (!req.session.user)` instead of explicitly checking if it is `undefined`.